File: daten-scripten/unterrichts-durchschnitt-wk.py
import pandas as pd

def main():
    schulen_path = "../daten-generator/daten/Schulen.xlsx"
    unterrichte_path = "../daten-generator/daten/Unterrichte.xlsx"

    schulen_df = pd.read_excel(schulen_path)
    unterrichte_df = pd.read_excel(unterrichte_path)

    wk_code = 644000000000

    filtered_schulen = schulen_df[
        (schulen_df["di_LandkreisKz"] == wk_code) &
        (schulen_df["di_Rechtsstellung"] == "ÖFF")
    ]

    unterrichte_filtered = unterrichte_df[
        unterrichte_df["un_DienststellenNr"].isin(filtered_schulen["di_DienststellenNr"])
    ]

    merged = pd.merge(
        unterrichte_filtered,
        filtered_schulen[["di_DienststellenNr", "di_Name"]],
        left_on="un_DienststellenNr",
        right_on="di_DienststellenNr",
        how="left"
    )

    grouped = merged.groupby("di_Name")["un_AnzahlSchueler"].agg(["mean", "median"]).reset_index()
    grouped.columns = ["Schulname", "Durchschnitt Schüler pro Unterricht", "Median Schüler pro Unterricht"]

    print("Mittlere und durchschnittliche Anzahl von Schülern im Unterricht (pro Schule im Wetteraukreis):\n")
    print(grouped)

if __name__ == "__main__":
    main()

File: r-daten-scripten/unterrichts-durchschnitt-wk.R
# ------------------------------------------------------------------------------
# Skript zur Auswertung der Unterrichtsdaten in öffentlichen Schulen im Wetteraukreis
# Ziel: Durchschnitt und Median der Schüleranzahl pro Unterricht und Schule
# Verwendet: Schulen.xlsx und Unterrichte.xlsx
# ------------------------------------------------------------------------------
library(readxl)  # zum Einlesen von Excel-Dateien
library(dplyr)   # für Datenverarbeitung

# Konstanten
source("common/konstanten.R")

# Schuldaten laden
schulen_df <- read_excel(schulen_path)

# Unterrichtsdaten laden
unterrichte_df <- read_excel(unterrichte_path)

# Öffentliche Schulen im Wetteraukreis filtern
filtered_schulen <- schulen_df %>%
  filter(
    di_LandkreisKz == wk_code,
    di_Rechtsstellung == "ÖFF"
  )

# Nur Unterrichte aus diesen Schulen auswählen
unterrichte_filtered <- unterrichte_df %>%
  filter(un_DienststellenNr %in% filtered_schulen$di_DienststellenNr)

# Schuldaten (Name) hinzufügen
merged <- unterrichte_filtered %>%
  left_join(
    filtered_schulen %>% select(di_DienststellenNr, di_Name),
    by = c("un_DienststellenNr" = "di_DienststellenNr")
  )

# Gruppierung nach Schulname und Berechnung von Mittelwert und Median
grouped <- merged %>%
  group_by(di_Name) %>%
  summarise(
    `Durchschnitt Schüler pro Unterricht` = mean(un_AnzahlSchueler, na.rm = TRUE),
    `Median Schüler pro Unterricht` = median(un_AnzahlSchueler, na.rm = TRUE),
    .groups = "drop"
  ) %>%
  rename(Schulname = di_Name)

# Ergebnis ausgeben
cat("Mittlere und durchschnittliche Anzahl von Schülern im Unterricht (pro Schule im Wetteraukreis):\n\n")
print(grouped)
